crds/operators.coreos.com_operatorconditions.yaml

```diff
@@ -0,0 +1,190 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.3.0
+  creationTimestamp: null
+  name: operatorconditions.operators.coreos.com
+spec:
+  group: operators.coreos.com
+  names:
+    kind: OperatorCondition
+    listKind: OperatorConditionList
+    plural: operatorconditions
+    singular: operatorcondition
+  scope: Namespaced
```

Comment:
I wonder if we consider this CRD to be cluster scoped? It will align with Operator API that way as well. Maybe that is not the intention here. Just curious.

Reply:
Thanks for bringing this up @dinhxuanvu, I had thought about this as well and I would love a sanity check on my reasoning.

Currently OLM allows multiple installs of the same operator provided the scopes do not intercept. I don't believe that one scoped operator should prevent a different scoped operator to upgrade. By having the CR namespaced, we can continue to support scoped upgrades.

If OLM ever deprecates OperatorGroups and all operator are global, then the namespaced operatorCondition could still be placed in the namespace that the operator pod is deployed to.

Reply:
I suppose we could accomplish the same behavior by using lables and associating the OperatorCondition with a specific operator, I would love to hear if @kevinrizza and @ecordell have a preference.

Reply:
>I suppose if we wanted the resource to be cluster scoped we could accomplish the same behavior by using lables and associating the OperatorCondition with a specific operator, I would love to hear if @kevinrizza and @ecordell have a preference.

The only negative experience here is that scoped operators now require the ability to write to a non-namespaced resource, which might ruffle some feathers.

Reply:
I can see your reasoning there. At the moment, operator is pretty much namespaced so it makes sense to have namspaced conditions. Operator API can aggregate Condition status just like any other OLM objects so it is fine to go this way. However, I thought there is a synergy between OperatorCondition and Operator API. I think cluster-scoped OperatorCondition can be used as a signal for an operator to know if the other operator that it may depend on is available or not. It can be useful that way. I'm not 100% sure though. Anyway, I certainly don't want to cause a scope creep with this comment.